src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2019,19 +2022,13 @@ private void GetSlotsToMarkAsNotNullable(BoundExpression operand, ArrayBuilder<i
                         slot = MakeSlot(operand);
                         if (slot > 0 && PossiblyNullableType(operand.Type))
                         {
-                            // If we got a slot then all previous BoundCondtionalReceivers must have been handled.
-                            Debug.Assert(_lastConditionalAccessSlot == -1);
-
                             slotBuilder.Add(slot);
                         }
 
                         break;
                 }
 
-                // If we didn't get a slot, it's possible that the current _lastConditionalSlot was never processed,
-                // so we reset before leaving the function.
-                _lastConditionalAccessSlot = -1;
-
+                _lastConditionalAccessSlot = previousConditionalAccessSlot;
```

Comment:
Did you consider a `finally` block here? Whenever we have resets like this I worry about the risk of a later change inserting a `return` in the middle. I may be over paranoid here.

Reply:
I don't think we need a `try`/`finally` here because any exception will terminate the analysis, so the state we leave it in won't matter.
We have very limited exception handling in flow analysis (only one `finally` to free the pooled object when hitting stackguard).

Reply:
The concern is with returning early rather than exceptions. I think I'd prefer to have this reset in a `finally` block.

---
In reply to: [264336845](https://github.com/dotnet/roslyn/pull/33997#discussion_r264336845) [](ancestors = 264336845)

Reply:
Sorry, I'd misunderstood. Will fix